### ISSUE TITLE
Query without pivot

### ISSUE THREAD

**Issue by garcipat** [influxdata/influxdb-client-csharp#279]
I wanted to query my data with a selection of fields. So I have fields for a various data and my application wants to query different fields at different times. So using pivot to move query them as columns doesnt work for me since they are not the same with every query.

To describe it a bit with code:
I have the class

And query it like:

```csharp

internal InfluxDBQueryable<DataPoint> CreateDataPointQuery<T>(QueryOptions queryOptions, DateTime start, DateTime end)
{
    var queryable = InfluxDBQueryable<DataPoint>.Queryable(
        _settings.BucketName,
        _settings.OrgId,
        _queryApi)
        .Where(dp => dp.TimeStamp >= start && dp.TimeStamp <= end);

    // Measurement
    if (queryOptions.Measurement.IsNotEmpty())
        queryable = queryable.Where(f => f.Measurement == queryOptions.Measurement);

    // Fields
    if (queryOptions.Fields.Any())
    {
        queryable = queryable.Where(f => queryOptions.Fields.Contains(f.Field));
    }

    return queryable.ToInfluxQueryable();
}

public class DataPoint
{
    [Column("_field")]
    public string Field { get; set; }

    [Column("_value")]
    public object Value { get; set; }

    [Column("_measurement", IsMeasurement = true)]
    public string Measurement { get; set; }

    [Column(IsTimestamp = true)]
    public DateTime TimeStamp { get; set; }
}
```

But since pivot is included automaticly, I cannot map my class with "_field" and "_value"

__Proposal:__
Allow deactivating pivot in query settings.

__Current behavior:__
Class cannot be mapped.

__Desired behavior:__
Allow mapping queries without pivot into class.

__Alternatives considered:__
Open for other solutions

__Use case:__
Well I cannot use Linq with this pivot. I have to create the query with strings. I need to have a list of datapoints without grouping them by timestamp

**Comment by bednar** [comment]
Hi @garcipat,

thanks for your suggestion. 

Is this something you might be willing to help with?

The PR is pretty straightforward... we have to add new option into https://github.com/influxdata/influxdb-client-csharp/blob/1d5db8681067996ca8def615278640a74236f304/Client.Linq/InfluxDBQueryable.cs#L16 and correctly modify https://github.com/influxdata/influxdb-client-csharp/blob/1b8eccb73bca11d23ff55fc51f408c853e55a6d4/Client.Linq/Internal/QueryAggregator.cs#L158

Regards

**Comment by garcipat** [comment]
> Hi @garcipat,
> 
> thanks for your suggestion.
> 
> Is this something you might be willing to help with?
> 
> The PR is pretty straightforward... we have to add new option into
> 
> https://github.com/influxdata/influxdb-client-csharp/blob/1d5db8681067996ca8def615278640a74236f304/Client.Linq/InfluxDBQueryable.cs#L16
> 
> and correctly modify
> https://github.com/influxdata/influxdb-client-csharp/blob/1b8eccb73bca11d23ff55fc51f408c853e55a6d4/Client.Linq/Internal/QueryAggregator.cs#L158
> 
> Regards

That would be it yes, to skip that pivot line if that property is set. Thats what my suggestion would be.
I dont know if there is a better solution to this, but that would be the easiest.

**Comment by garcipat** [comment]
If the idea is that I do a PR myself, I tried to create a feature branch, but couldnt push it. Sorry first time actually contributing within github. Dont know If Im doing something wrong.

**Comment by bednar** [comment]
You have create a fork of repository and push your changes into this fork.

**Comment by garcipat** [comment]
Okay cool. Im trying to add a test to test the addition I made. But cant get the influxdb configured correctly that the test can access my docker container. Adjusting bucket, orgId and token doesnt seem enough. @bednar are there some instructions how to get the test environment going? Would be quiet helpful. You can set a default user but I dont know what user is used for the tests and couldnt discover that so far.

**Comment by bednar** [comment]
Thanks for your participation.

> Edit: just discovered the scripts for setting up the docker container.

Just fyi... the `/Scripts/influxdb-restart.sh` setups docker instance configured for testing.

**Comment by bednar** [comment]
The issue is fixed in [4.0.0-rc2](https://github.com/influxdata/influxdb-client-csharp/pull/291) milestone. You can use latest dev version by:

```console
dotnet add package InfluxDB.Client --version 4.0.0-dev.5460
```

If you would like to disable appending `pivot()` function into produced Flux query, you can use:

```c#
var optimizerSettings =
    new QueryableOptimizerSettings
    {
        AlignFieldsWithPivot = false
    };
    
var query = from s in InfluxDBQueryable<Sensor>.Queryable("my-bucket", "my-org", queryApi, optimizerSettings)
    select s;
```

**Comment by garcipat** [comment]
Cool. Was trying to get the environment going but couldnt. And didnt have
enough time at work to do it. it was lower priorized.

Jakub Bednář ***@***.***> schrieb am Mi., 23. Feb. 2022,
08:00:
